File: Java/Creational/Simple-Factory/Examples/Cake/CakeFactory.java
public class CakeFactory {
    public static Cake makeCake(Flavour flavour, Shape shape) {
        switch (flavour) {
            case CHOCOLATE:
                return new ChocolateCake(shape);
            case VANILLA:
                return new VanillaCake(shape);
            default:
                return null;
        }
    }
}

File: Typescript/Creational/Simple-Factory/Examples/Cake/CakeFactory.ts
import { Flavour, Shape } from "./Cake"
import { ChocolateCake } from "./ChocolateCake"
import { VanillaCake } from "./VanillaCake"

export class CakeFactory {
	static makeCake = (cakeFlavour: Flavour, cakeShape: Shape) => {
		switch (cakeFlavour) {
			case Flavour.Chocolate:
				return new ChocolateCake(cakeShape)
			case Flavour.Vanilla:
				return new VanillaCake(cakeShape)
		}
	}
}
